src/parser/shared/normalizers/PrePullCooldowns.js

```diff
@@ -109,7 +108,11 @@ class PrePullCooldowns extends EventsNormalizer {
         for (let i = 0; i < buffSpells.length; i += 1) {
           if (buffSpells[i].buffId === event.ability.guid) {
             debug && console.debug(`Detected a precast buff cooldown: ${event.ability.name}`);
-            prepullCasts.push(this.constructor._fabricateCastEvent(event, buffSpells[i].castId));
+            if(buffSpells[i].castId instanceof Array && buffSpells[i].castId.includes(event.ability.guid)){ //try to find corresponding cast, otherwise pass list of casts
+              prepullCasts.push(this.constructor._fabricateCastEvent(event));
```

Comment:
shouldn't this pass `event.ability.guid` as second param?

Reply:
without a param it defaults to event.ability as the new event.ability so the guid will be copied over too